website/pages/docs/internals/filesystem.mdx

```diff
@@ -0,0 +1,464 @@
+drwxrwxrwx  4.0 KiB  2020-10-27T18:51:54Z  alloc/
+drwxrwxrwx  4.0 KiB  2020-10-27T18:51:54Z  task1/
+
+$ nomad alloc fs b0686b27 task1
+Mode         Size     Modified Time         Name
+drwxrwxrwx   4.0 KiB  2020-10-27T18:51:54Z  local/
+drwxrwxrwx   60 B     2020-10-27T18:51:54Z  secrets/
+dtrwxrwxrwx  4.0 KiB  2020-10-27T18:51:54Z  tmp/
+
+$ nomad alloc fs b0686b27 task1/local
+Mode  Size  Modified Time  Name
+```
+
+If you inspect the Docker container that's created, you'll see three
+directories bind-mounted into the container:
+
+```shell-session
+$ docker inspect 32e | jq '.[0].HostConfig.Binds'
+[
+  "/var/nomad/alloc/b0686b27-8af3-8252-028f-af485c81a8b3/alloc:/alloc",
+  "/var/nomad/alloc/b0686b27-8af3-8252-028f-af485c81a8b3/task1/local:/local",
+  "/var/nomad/alloc/b0686b27-8af3-8252-028f-af485c81a8b3/task1/secrets:/secrets"
+]
+```
+
+The root filesystem inside the container can see these three mounts, along
+with the rest of the container filesystem:
+
+```shell-session
+$ docker exec -it 32e /bin/sh
+# ls /
+alloc  boot  dev  home  lib64  media  opt   root  sbin     srv  tmp  var
+bin    data  etc  lib   local  mnt    proc  run   secrets  sys  usr
+```
+
+Note that because the three directories are bind-mounted into the container
+filesystem, nothing written outside those three directories elsewhere in the
+allocation working directory will be accessible inside the container. This
+means templates, artifacts, and dispatch payloads for tasks with `image`
+isolation must be written into the `NOMAD_ALLOC_DIR`, `NOMAD_TASK_DIR`, or
+`NOMAD_SECRETS_DIR`.
+
+To work around this limitation, you can use the task driver's mounting
+capabilities to mount one of the three directories to another location in the
+task. For example, with the Docker driver you can use the driver's `mounts`
+block to bind a secret written by a `template` block to the
+`NOMAD_SECRETS_DIR` into a configuration directory elsewhere in the task:
+
+```hcl
+job "example" {
+  datacenters = ["dc1"]
+
+  task "task1" {
+    driver = "docker"
+
+    config {
+      image = "redis:6.0"
+      mounts = [{
+        type     = "bind"
+        source   = "secrets"
+        target   = "/etc/redis.d"
+        readonly = true
+      }]
+
+      template {
+        destination = "${NOMAD_SECRETS_DIR}/redis.conf"
+        data        = <<EOT
+{{ with secret "secrets/data/redispass" }}
+requirepass {{- .Data.data.passwd -}}{{end}}
+EOT
+
+      }
+    }
+  }
+}
+```
+
+
+### `chroot` isolation
+
+Task drivers like `exec` or `java` (on Linux) use `chroot` isolation, where
+the task driver isolates task filesystems with `chroot` or `pivot_root`. These
+isolated filesystems will be built inside the task working directory.
+
+You can see an example of `chroot` isolation by running the following minimal
+job on Linux:
+
+```hcl
+job "example" {
+  datacenters = ["dc1"]
+
+  task "task1" {
+    driver = "exec"
+
+    config {
+      image = "redis:6.0"
+    }
+  }
```

Comment:
Looks like maybe some copy paste from docker.  I extrapolated based on the raw_exec job for the suggestion.

```suggestion
  task "task2" {
    driver = "exec"

    config {
      command = "/bin/sh"
      args = ["-c", "sleep 600"]
    }
  }
```